src/Esfa.Vacancy.Register.Infrastructure/Repositories/VacancyRepository.cs

```diff
@@ -1,13 +1,74 @@
-﻿using System.Threading.Tasks;
+﻿using Dapper;
+using Esfa.Vacancy.Register.Domain.Repositories;
+using Esfa.Vacancy.Register.Infrastructure.Settings;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using DomainEntities = Esfa.Vacancy.Register.Domain.Entities;
 
 namespace Esfa.Vacancy.Register.Infrastructure.Repositories
 {
     public class VacancyRepository : IVacancyRepository
     {
-        public async Task<Domain.Entities.Vacancy> GetVacancyByReferenceNumberAsync(int referenceNumber)
+        private readonly IProvideSettings _provideSettings;
+
+        public VacancyRepository(IProvideSettings provideSettings)
+        {
+            _provideSettings = provideSettings;
+        }
+
+        public async Task<DomainEntities.Vacancy> GetVacancyByReferenceNumberAsync(int referenceNumber)
         {
-            await Task.Delay(10);
-            return null;
+            var connectionString =
+                _provideSettings.GetSetting(ApplicationSettingConstants.AvmsPlusDatabaseConnectionStringKey);
```

Comment:
By using _provideSettings.GetSetting_ you end up dictating that the settings provider has to be dictionary. You did a good job by abstracting the key with _ApplicationSettingConstants.AvmsPlusDatabaseConnectionStringKey_

Alteratively you could abstract this into an settings class so that it's strongly typed and could make testing a little easier
## ApplicationSettings.cs
```c#
namespace Esfa.Vacancy.Register.Infrastructure.Settings
{
    public sealed class ApplicationSettings : IApplicationSettings
    {
        private readonly IProvideSettings _settings;

        public ApplicationSettings(IProvideSettings settings)
        {
            _settings = settings;
        }

        public string SomeSetting => _settings.GetSetting("SomeSetting");
    }
}
```